samza-core/src/main/java/org/apache/samza/operators/triggers/RepeatingTriggerImpl.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.operators.triggers;
+
+import org.apache.samza.operators.data.MessageEnvelope;
+import org.apache.samza.util.Clock;
+
+/**
+ * Implementation class for a {@link RepeatingTrigger}
+ */
+public class RepeatingTriggerImpl<M extends MessageEnvelope> implements TriggerImpl<M> {
+
+  private final Trigger<M> repeatingTrigger;
+  private final Clock clock;
+
+  private TriggerImpl<M> currentTriggerImpl;
+  private boolean shouldFire = false;
+
+  public RepeatingTriggerImpl(RepeatingTrigger<M> repeatingTrigger, Clock clock) {
+    this.repeatingTrigger = repeatingTrigger.getTrigger();
+    this.clock = clock;
+    this.currentTriggerImpl = TriggerImpls.createTriggerImpl(this.repeatingTrigger, clock);
+  }
+
+  @Override
+  public void onMessage(M message, TriggerContext context) {
+    currentTriggerImpl.onMessage(message, context);
+  }
+
+  @Override
+  public void cancel() {
+    currentTriggerImpl.cancel();
+  }
+
+  @Override
+  public void clear() {
```

Comment:
No strong preference on this one. Re-wrote as-per  your suggestion.

>>It feels slightly weird that this clear method exists on the Trigger interface solely for RepeatingTrigger.

- For instance, other APIs (like beam) have a `onTrigger` method that `RepeatingTriggerImpls` can use to potentially reset themselves. 

- It was not clear if other trigger implementations will potentially need to clear their state.

Removing it now, and let's plan to add it later (if there are use-cases). The `instanceOf` checks are not pretty though.